[v0.6.3]windows d2.exe -w is crashed.

>d2.exe -w "sandbox.d2" "sandbox.svg"

'detected change' occurs frequently.
i did not save 'sandbox.d2'. (only the first time)
'd2.exe' sometimes crashed.

```
[34minfo[0m: broadcasting update to 1 client
[34minfo[0m: detected change in sandbox.svg: recompiling...
[32msuccess[0m: successfully compiled sandbox.d2 to sandbox.svg in 103.344ms
[34minfo[0m: broadcasting update to 1 client
[34minfo[0m: detected change in sandbox.svg: recompiling...
[32msuccess[0m: successfully compiled sandbox.d2 to sandbox.svg in 125.8546ms
[34minfo[0m: broadcasting update to 1 client
[34minfo[0m: detected change in sandbox.svg: recompiling...
[32msuccess[0m: successfully compiled sandbox.d2 to sandbox.svg in 95.0012ms
[34minfo[0m: broadcasting update to 1 client
[34minfo[0m: detected change in sandbox.svg: recompiling...
[32msuccess[0m: successfully compiled sandbox.d2 to sandbox.svg in 107.1851ms
[34minfo[0m: broadcasting update to 1 client
[34minfo[0m: detected change in sandbox.svg: recompiling...
[32msuccess[0m: successfully compiled sandbox.d2 to sandbox.svg in 102.4472ms
[34minfo[0m: broadcasting update to 1 client
[34minfo[0m: detected change in sandbox.svg: recompiling...
[32msuccess[0m: successfully compiled sandbox.d2 to sandbox.svg in 147.4784ms
[34minfo[0m: broadcasting update to 1 client
[34minfo[0m: detected change in sandbox.svg: recompiling...
[32msuccess[0m: successfully compiled sandbox.d2 to sandbox.svg in 88.3592ms
[34minfo[0m: broadcasting update to 1 client
[34minfo[0m: detected change in sandbox.svg: recompiling...
[32msuccess[0m: successfully compiled sandbox.d2 to sandbox.svg in 153.6542ms
[34minfo[0m: broadcasting update to 1 client
[34minfo[0m: detected change in sandbox.svg: recompiling...
[32msuccess[0m: successfully compiled sandbox.d2 to sandbox.svg in 87.7296ms
[34minfo[0m: broadcasting update to 1 client
[34minfo[0m: detected change in sandbox.svg: recompiling...
.
.
.
.
.
.
panic: runtime error: index out of range [0] with length 0

goroutine 13 [running]:
oss.terrastruct.com/d2/d2cli.(*watcher).watchLoop(0xc00007a1e0, {0x282ff88, 0xc000707c70})
        oss.terrastruct.com/d2/d2cli/watch.go:297 +0xb7e
oss.terrastruct.com/d2/d2cli.(*watcher).goFunc.func1()
        oss.terrastruct.com/d2/d2cli/watch.go:203 +0x76
created by oss.terrastruct.com/d2/d2cli.(*watcher).goFunc
        oss.terrastruct.com/d2/d2cli/watch.go:199 +0x8d
```

```
[34minfo[0m: detected change in sandbox.svg: recompiling...
[32msuccess[0m: successfully compiled sandbox.d2 to sandbox.svg in 507.888ms
[34minfo[0m: broadcasting update to 1 client
[34minfo[0m: detected change in sandbox.svg: recompiling...
[32msuccess[0m: successfully compiled sandbox.d2 to sandbox.svg in 582.9534ms
[34minfo[0m: broadcasting update to 1 client
[34minfo[0m: detected change in sandbox.svg: recompiling...
panic: runtime error: index out of range [0] with length 0

goroutine 7 [running]:
oss.terrastruct.com/d2/d2cli.(*watcher).watchLoop(0xc0005d8000, {0x1e3ff88, 0xc0000d2050})
        oss.terrastruct.com/d2/d2cli/watch.go:297 +0xb7e
oss.terrastruct.com/d2/d2cli.(*watcher).goFunc.func1()
        oss.terrastruct.com/d2/d2cli/watch.go:203 +0x76
created by oss.terrastruct.com/d2/d2cli.(*watcher).goFunc
        oss.terrastruct.com/d2/d2cli/watch.go:199 +0x8d
```

--- a/d2cli/watch.go
+++ b/d2cli/watch.go
@@ -292,15 +292,15 @@ func (w *watcher) watchLoop(ctx context.Context) error {
 			var changedList []string
 			for k := range changed {
 				changedList = append(changedList, k)
+				delete(changed, k)
 			}
 			sort.Strings(changedList)
 			changedStr := w.ms.HumanPath(changedList[0])
-			for i := 1; i < len(changed); i++ {
+			for i := 1; i < len(changedList); i++ {
 				changedStr += fmt.Sprintf(", %s", w.ms.HumanPath(changedList[i]))
 			}
 			w.ms.Log.Info.Printf("detected change in %s: recompiling...", changedStr)
 			w.requestCompile()
-			changed = make(map[string]struct{})
 		case err, ok := <-w.fw.Errors:
 			if !ok {
 				return errors.New("fsnotify watcher closed")

